prog/sensors/chips.c

```diff
@@ -67,6 +67,53 @@ void print_chip_raw(const sensors_chip_name *name)
 	}
 }
 
+void print_chip_json(const sensors_chip_name *name)
+{
+	int a, b, cnt, subCnt, err;
+	const sensors_feature *feature;
+	const sensors_subfeature *sub;
+	char *label;
+	double val;
+
+	a = 0;
+	cnt = 0;
+	while ((feature = sensors_get_features(name, &a))) {
+		if (!(label = sensors_get_label(name, feature))) {
+			fprintf(stderr, "ERROR: Can't get label of feature "
+				"%s!\n", feature->name);
+			continue;
+		}
+		if (cnt > 0)
+			printf(",\n", label);
+		printf("      \"%s\":{\n", label);
+		free(label);
+
+		b = 0;
+		subCnt = 0;
+		while ((sub = sensors_get_all_subfeatures(name, feature, &b))) {
+			if (sub->flags & SENSORS_MODE_R) {
+				if ((err = sensors_get_value(name, sub->number,
+							     &val)))
+					fprintf(stderr, "ERROR: Can't get "
+						"value of subfeature %s: %s\n",
+						sub->name,
+						sensors_strerror(err));
+				else {
+					if (subCnt > 0)
+						printf(",\n", label);
+					printf("         \"%s\": %.3f", sub->name, val);
+				}
+
+			} else
+			printf("(%s)", label);
+			subCnt++;
+		}
+		printf("\n      }", label);
```

Comment:
Not sure what you want to do here (print the label or not ?), but as written this is wrong (and gives you a warning when compiling).